reactive/kotlinx-coroutines-rx2/src/RxConvert.kt

```diff
@@ -77,6 +81,34 @@ public fun <T : Any> ReceiveChannel<T>.asObservable(context: CoroutineContext):
         send(t)
 }
 
+/**
+ * Transforms given cold [ObservableSource] into cold [Flow].
+ *
+ * The resulting flow is _cold_, which means that [ObservableSource.subscribe] is called every time a terminal operator
+ * is applied to the resulting flow.
+ *
+ * A channel with the [default][Channel.BUFFERED] buffer size is used. Use the [buffer] operator on the
+ * resulting flow to specify a user-defined value and to control what happens when data is produced faster
+ * than consumed, i.e. to control the back-pressure behavior. Check [callbackFlow] for more details.
+ */
+@ExperimentalCoroutinesApi
+public fun <T: Any> ObservableSource<T>.asFlow(): Flow<T> = callbackFlow {
+
+    var disposable = Disposables.empty()
+
+    val observer = object : Observer<T> {
+        override fun onComplete() { close() }
+        override fun onSubscribe(d: Disposable) = if (disposable.isDisposed) d.dispose() else disposable = d
```

Comment:
There is still a race between the write of `var disposable` field by this `onSubscribe` method and its read in `awaitClose` lambda. You need to atomic reference here just like Rx does in its operator implementations.